### PR TITLE
Feature/deserializer

### PR DESCRIPTION
**Pull request by Le-Caignec**

### CODE REVIEW

File: tools/dataProtector/advanced/iDapp/deserializer.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+[Borsch technical specification](https://borsh.io/). We developed this JavaScript
+library to simplify deserialization in your iDapp built with JavaScript.
+
+::: warning
+
+If you want to build your iDapp in another language, you should follow the Borsch
+serialization specification and reimplement it in your chosen language such as
+Python, Rust, Go, etc.
+
+:::
```
Comment (PierreJeanjacquot) [inline tools/dataProtector/advanced/iDapp/deserializer.md:25]:
```suggestion
::: warning

If you want to build your iDapp in another language, you need how to deserialize a protected data.

Under the hood, protected data are zip files replicating the tree structure of the original data object.
Each value is stored in a dedicated file, binary values are stored as is while boolean, numbers, and strings
are serialized with borsh.

To access a value from a protected data, your app will need to unzip the iExec dataset file at
`$IEXEC_IN/$IEXEC_DATASET_FILENAME` then for `'bool'`, `'f64'`, `'i128'`, or `'string'` types use the
Borsch deserialization specification to recover the original value.
Borsh has [implementations in various languages](https://github.com/near/borsh#implementations), check
your favorite one.

:::
```

File: tools/dataProtector/advanced/iDapp/deserializer/getValue.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+const value1 = await deserializer.getValue(
+    'path.to.value1', 
+    'bool' // [!code focus]
+    );
+```
+<!-- prettier-ignore-end -->
+
+## Return value
+
+The deserialize file.
```
Comment (PierreJeanjacquot) [inline tools/dataProtector/advanced/iDapp/deserializer/getValue.md:68]:
```suggestion
The recovered original value.
```

File: tools/dataProtector/advanced/iDapp/deserializer/getValue.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+### file
+
+`BooleanSchemaFilter<T>     | NumberSchemaFilter<T>     | BigintSchemaFilter<T>     | StringSchemaFilter<T>     | BinarySchemaFilter<T>`
```
Comment (PierreJeanjacquot) [inline tools/dataProtector/advanced/iDapp/deserializer/getValue.md:72]: remove this

```suggestion
```

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]
see comments

File: tools/dataProtector/advanced/iDapp.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# iDapp
```
[the remark this answers is not in the document]
Comment (Le-Caignec) [reply tools/dataProtector/advanced/iDapp.md:1]: yeah, the topic has been taken by jean

**Pull request merged**